.baseline/checkstyle/checkstyle.xml

```diff
@@ -165,6 +165,11 @@
             <property name="illegalClasses" value="com.google.common.base.Optional, com.google.common.base.Supplier"/>
             <message key="import.illegal" value="Use the Java8 version of Guava objects."/>
         </module>
+        <module name="IllegalImport">
+            <property name="id" value="BanUnrelocatedGuavaClasses"/>
```

Comment:
Hmmm, the intention with the original PR is that the classpath would be set up in such a way that the other subprojects would not be able to see Guava on their classpath and these imports would give compile errors. @rdblue I guess you didn't manage to get that working? Perhaps #1067 will fix this? So this rule shouldn't be necessary and compile errors would give devs a warning earlier than checkstyle errors. I don't really mind having this as an extra safety net though.